src/contracts/AbstractARM.sol

```diff
@@ -0,0 +1,714 @@
+    address internal constant DEAD_ACCOUNT = 0x000000000000000000000000000000000000dEaD;
+    /// @notice The scale of the performance fee
+    /// 10,000 = 100% performance fee
+    uint256 public constant FEE_SCALE = 10000;
+
+    ////////////////////////////////////////////////////
+    ///             Immutable Variables
+    ////////////////////////////////////////////////////
+
+    /// @notice The address of the asset that is used to add and remove liquidity. eg WETH
+    address public immutable liquidityAsset;
+    /// @notice The asset being purchased by the ARM and put in the withdrawal queue. eg stETH
+    address public immutable baseAsset;
+    /// @notice The swap input token that is transferred to this contract.
+    /// From a User perspective, this is the token being sold.
+    /// token0 is also compatible with the Uniswap V2 Router interface.
+    IERC20 public immutable token0;
+    /// @notice The swap output token that is transferred from this contract.
+    /// From a User perspective, this is the token being bought.
+    /// token1 is also compatible with the Uniswap V2 Router interface.
+    IERC20 public immutable token1;
+    /// @notice The delay before a withdrawal request can be claimed in seconds. eg 600 is 10 minutes.
+    uint256 public immutable claimDelay;
+
+    ////////////////////////////////////////////////////
+    ///             Storage Variables
+    ////////////////////////////////////////////////////
+
+    /**
+     * @notice For one `token0` from a Trader, how many `token1` does the pool send.
+     * For example, if `token0` is WETH and `token1` is stETH then
+     * `traderate0` is the WETH/stETH price.
+     * From a Trader's perspective, this is the buy price.
+     * From the ARM's perspective, this is the sell price.
+     * Rate is to 36 decimals (1e36).
+     * To convert to a stETH/WETH price, use `PRICE_SCALE * PRICE_SCALE / traderate0`.
+     */
+    uint256 public traderate0;
+    /**
+     * @notice For one `token1` from a Trader, how many `token0` does the pool send.
+     * For example, if `token0` is WETH and `token1` is stETH then
+     * `traderate1` is the stETH/WETH price.
+     * From a Trader's perspective, this is the sell price.
+     * From a ARM's perspective, this is the buy price.
+     * Rate is to 36 decimals (1e36).
+     */
+    uint256 public traderate1;
+    /// @notice The price that buy and sell prices can not cross scaled to 36 decimals.
+    /// This is also the price the base assets, eg stETH, in the ARM contract are priced at in `totalAssets`.
+    uint256 public crossPrice;
+
+    /// @notice Cumulative total of all withdrawal requests including the ones that have already been claimed.
+    uint120 public withdrawsQueued;
+    /// @notice Total of all the withdrawal requests that have been claimed.
+    uint120 public withdrawsClaimed;
+    /// @notice Index of the next withdrawal request starting at 0.
+    uint16 public nextWithdrawalIndex;
```

Comment:
🔴 I'm concerned about this id number only being only 16 bits. That's a lifetime max of 65K redeems. It's only read and written when creating a new withdraw request, so gas is not really a concern here.

What if we just make this a `uint256`, give it own storage slot, and call it a day. `withdrawsClaimed` and `withdrawsQueued` will bump up to 128 then.

Reply:
Fair enough. The LP functions are not gas critical so we can use an extra slot for long term peace of mind

Reply:
This change has been done